cinder/autogenerated_client.go

```diff
@@ -0,0 +1,1821 @@
+package cinder
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+type RequestHandlerFn func(*http.Request) (*http.Response, error)
+
+type UpdateVolumeTypeParams struct {
+
+	// VolumeType is required.
+	//
+	// A volume type offers a way to categorize or group volumes.
+	VolumeType string `json:"volume_type"`
+
+	//
+	// The unique identifier of the tenant or account.
+	TenantId string `json:"-"`
+
+	// VolumeTypeId is required.
+	//
+	// The unique identifier for an existing volume type.
+	VolumeTypeId string `json:"-"`
+}
+
+type UpdateVolumeTypeResults struct {
+	VolumeType struct {
+		ExtraSpecs struct {
+			Capabilities string `json:"capabilities"`
+		} `json:"extra_specs"`
+		ID   string `json:"id"`
+		Name string `json:"name"`
+	} `json:"volume_type"`
+}
+
+//
+// Updates a volume type.
+func updateVolumeType(request RequestHandlerFn, args UpdateVolumeTypeParams) (*UpdateVolumeTypeResults, error) {
```

Comment:
I wonder if these shouldn't be exported. We might periodically regenerate these, but not add exported methods on Client for every function. Then, others can make use of these auto-generated functions without having to go through the process of adding an exported method, writing tests, etc.